cryptonote_utils/cryptonote_utils.js

```diff
@@ -1670,7 +1679,23 @@ var cnUtil = function(currencyConfig)
 				throw "dst.amount < 0"; //amount can be zero if no change
 			}
 			dsts[i].keys = this.decode_address(dsts[i].address);
-			var out_derivation = this.generate_key_derivation(dsts[i].keys.view, txkey.sec);
+			
+			var out_derivation;
+			
+			// R = rD for subaddresses
+			if(this.is_subaddress(dsts[i].address)) {
+				txkey.pub = ge_scalarmult(dsts[i].keys.spend, txkey.sec);
```

Comment:
Do we need to initialize `out_derivation` in this case?

Reply:
It would need to be initialized shortly down the line. Really it should be directly after rD is (potentially) calculated